aten/src/ATen/native/cudnn/MHA.cpp

```diff
@@ -553,7 +579,8 @@ void run_cudnn_SDP_fprop(
     Tensor& dropoutoffset) {
   cudnnHandle_t handle = getCudnnHandle();
   if (!o.defined()) {
-    o = at::empty({b, h, s_q, d_v}, q.options());
+    // q is passed to us in BHSD dim order
+    alloc_with_matching_layout(q, o, {b, h, s_q, d_v});
```

Comment:
curious if cudnn can really handle output with arbitrary permutation, or is it only the contiguous and `transpose(1,2)` cases that are seen in the wild.

Reply:
I assume so (main restriction is in line with other backends: `check_last_dim_stride_equals_1_dense` but hard to say without basically fuzzing configurations